server/src/main/java/org/elasticsearch/search/internal/SearchContext.java

```diff
@@ -369,6 +370,31 @@ public Query rewrittenQuery() {
      */
     public abstract Profilers getProfilers();
 
+    /**
+     * The circuit breaker used to account for the search operation.
+     */
+    public abstract CircuitBreaker circuitBreaker();
+
+    /**
+     * Return the amount of memory to buffer locally before accounting for it in the breaker.
+     */
+    public abstract long memAccountingBufferSize();
```

Comment:
is it necessary to add these two new methods to SearchContext? I am not quite following where they are used, are they needed mostly for testing purposes or is there more to it?

Reply:
I guess what I am wondering is if we could make them perhaps arguments of the following method instead, or something along those lines, just to decrease the blast radius of this change.